File: banco-server/controllers/transactionController.js
// controllers/transactionController.js
const Transaction = require('../models/transactionModel');

const transactionController = {
  async listTransactions(req, res) {
    try {
      const { userId } = req.params;
      const transactions = await Transaction.findByUserId(userId);
      res.json(transactions);
    } catch (error) {
      console.error(error);
      res.status(500).json({ error: 'Error al obtener transacciones' });
    }
  }
};

module.exports = transactionController;

File: banca-front/lib/controllers/transaction_controller.dart
import 'package:flutter/material.dart';
import '../models/transaction.dart';
import '../services/api_service.dart';

class TransactionController with ChangeNotifier {
  final ApiService _apiService = ApiService();
  List<TransactionModel> transactions = [];

  Future<void> fetchTransactions(String userId, String idToken) async {
    try {
      final response = await _apiService.getTransactions(userId, idToken);
      transactions = response
          .map<TransactionModel>((json) => TransactionModel.fromJson(json))
          .toList();
      notifyListeners();
    } catch (e) {
      print('Error al obtener transacciones: $e');
    }
  }
}
